tests/python_package_test/test_plotting.py

```diff
@@ -21,6 +22,20 @@ def breast_cancer_split():
                             test_size=0.1, random_state=1)
 
 
+@pytest.fixture(scope="module")
+def categorical_data():
```

Comment:
What's the value of having a fixture return a function like this, that the tests then have to call? Unless there's something I'm not understanding, I think this is unnecessary complexity. Please just define a function `def _create_categorical_data` in this module instead (not a fixture), just as many of the other tests in this project are structured.